### PR TITLE
Modify integration configuration to explicitly use intel MPI and intel compiler

### PR DESCRIPTION
**Pull request by cmcantalupo**

### CODE REVIEW

File: integration/config/mcfly_env.sh [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+export MPICC=mpiicc
+export MPICXX=mpiicpc
+export FC=ifort
+export F77=ifort
+export F90=ifort
+export MPIFORT=mpiifort
+export MPIFC=mpiifort
+export MPIF77=mpiifort
+export MPIF90=mpiifort
+export FI_PROVIDER=psm2
```
Comment (bgeltz) [inline integration/config/mcfly_env.sh]:
```suggestion
```

Why is this necessary?

File: integration/config/mcfly_env.sh [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+#  (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY LOG OF THE USE
+#  OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+#
+
+# MCFLY BUILD ENVIRONMENT
+#
+# This script is intended to be sourced within an existing script or shell ONLY.
+# It is NOT intended to be ./ executed.
+
+module swap mvapich2 impi >& /dev/null || true
```
Comment (bgeltz) [inline integration/config/mcfly_env.sh]:
```suggestion
```

I'm going to change the cluster wide defaults if this is necessary.

**Review by bgeltz — CHANGES_REQUESTED** [review]
I thought these changes were going to go into build_env.sh.

When we wrote build_env.sh originally, it was written for the default McFly environment.

File: integration/config/mcfly_env.sh (hunk 2 above)
Comment (cmcantalupo) [reply integration/config/mcfly_env.sh]: Not the best solution as written anyway (no check for impi being loaded in the end)

**Review by dannosliwcd — APPROVED** [review]

**Review by bgeltz — APPROVED** [review]
